xds/src/main/java/io/grpc/xds/EnvoyProtoData.java

```diff
@@ -1113,6 +1138,270 @@ public String toString() {
     }
   }
 
+  static StructOrError<HttpFault> decodeFaultFilterConfig(Any rawFaultFilterConfig) {
+    if (rawFaultFilterConfig.getTypeUrl().equals(
+        "type.googleapis.com/envoy.config.filter.http.fault.v2.HTTPFault")) {
```

Comment:
nit: define constants for these type urls.

Reply:
Well, this is style preference :stuck_out_tongue:. If the constant is used only one place in the main code then I don't favor static constant as this is more straightforward.

Reply:
There is little to do with being used once or multiple times.

This string is long. Also more importantly, this is something coming from the xDS API definition, instead of something that the implementer comes up with. So declaring it as a constant on the top is much readable and understandable for readers.

Reply:
I disagree. Declaring it as a constant adds an extra indirection and it's less readable than put it inline. An inline constant is still a constant. There is little to do with the string being long or short, unless it has too many lines.

Reply:
> It makes your code more maintainable since it's easier to see what can be changed at a later time (if necessary).

If you read posts like [this](https://stackoverflow.com/questions/4234129/should-i-use-constants-instead-of-strings-even-if-the-strings-are-only-ever-used) and [this](https://softwareengineering.stackexchange.com/questions/145738/should-a-string-constant-be-defined-if-its-only-going-to-be-used-once). Most people would vote for declaring a String constant on the top.

Reply:
Still, I am not in favor of having these two String literals used here. It looks so messy. We've been doing the conventional way in XdsClient. This is really bad.